swan-lake/integration-technology-tools/edi-tool.md

```diff
@@ -6,4 +6,349 @@ keywords: ballerina, programming language, edi, contract
+    |--lib
+    |--schemas
+       |--850.json
+       |--810.json
+       |--820.json
+       |--855.json
+```
+
+Execute the `libgen` command to generate a Ballerina library as shown below.
+
+```
+bal edi libgen -O citymart -n porder -s CityMart/schemas -o CityMart/lib
+```
+
+The generated Ballerina library will be as shown below.
+
+```ballerina
+|-- CityMart
+    |--lib  
+    |--porder
+    |     |--modules
+    |	  |   |--m850
+    |	  |	  |  |--G_850.bal
+    |     |   |  |--transformer.bal
+    |	  |	  |--m810
+    |	  |	  |  |--G_810.bal
+    |     |   |  |--transformer.bal
+    |	  |	  |--m820
+    |	  |	  |  |--G_820.bal
+    |     |   |  |--transformer.bal
+    |	  |	  |--m855
+    |	  |	    |--G_855.bal
+    |     |     |--transformer.bal
+    |	  |--Ballerina.toml
+    |	  |--Module.md
+    |	  |--Package.md
+    |	  |--porder.bal
+    |	  |--rest_connector.bal
+    |
+    |--schemas
+       |--850.json
+       |--810.json
+       |--820.json
+       |--855.json
+```
+
+As seen in the above project structure, the code for each EDI schema is generated into a separate module to prevent possible conflicts. Now, it is possible to build the above project using the `bal pack` command and publish it into [Ballerina Central](https://central.ballerina.io/) using the `bal push` command. 
+
+Then, any Ballerina project can import this package and use it to work with the EDI files related to purchase orders. An example of using this library for reading an `850` file and writing an `855` file is shown below.
+
+```ballerina
+import ballerina/io;
+import citymart/porder.m850;
+import citymart/porder.m855;
+
+public function main() returns error? {
+    string orderText = check io:fileReadString("orders/d15_05_2023/order10.edi");
+    m850:Purchase_Order purchaseOrder = check m850:fromEdiString(orderText);
+    ...
+    m855:Purchase_Order_Acknowledgement orderAck = {...};
+    string orderAckText = check m855:toEdiString(orderAck);
+    check io:fileWriteString("acks/d15_05_2023/ack10.edi", orderAckText);
+}
+```
+
+It is quite common for different trading partners to use variations of the standard EDI formats. In such cases, it is possible to create partner-specific schemas and generate a partner-specific Ballerina library for processing interactions with a particular partner.
+
+### Use generated EDI libraries as standalone REST services
+
+The EDI libraries generated above can also be compiled into a JAR file (using the `bal build` command) and executed as a standalone Ballerina service that processes EDI files via a REST interface. This is useful for microservices environments where the EDI processing functionality can be deployed as a separate microservice.
+
+For example, the `citymart` library generated above can be built and executed as a JAR file. Once executed, it will expose a REST service to work with `X12 850`, `810`, `820`, and `855` files. 
+
+You can convert `X12 850` EDI text to JSON using the REST service as shown below.
```

Comment:
Missing comma.

Reply:
Also what is given below is not the REST service, but the curl command.

Reply:
Updated.